opentelemetry-instrumentation/src/opentelemetry/instrumentation/utils.py

```diff
@@ -152,3 +156,123 @@ def _python_path_without_directory(python_path, directory, path_separator):
         "",
         python_path,
     )
+
+
+_OTEL_SEMCONV_STABILITY_OPT_IN_KEY = "OTEL_SEMCONV_STABILITY_OPT_IN"
+
+# Old to new http semantic convention mappings
+_OTEL_HTTP_SEMCONV_MIGRATION_MAPPING = {
+    SpanAttributes.HTTP_METHOD: SpanAttributes.HTTP_REQUEST_METHOD,
+    SpanAttributes.HTTP_STATUS_CODE: SpanAttributes.HTTP_RESPONSE_STATUS_CODE,
+    SpanAttributes.HTTP_SCHEME: SpanAttributes.URL_SCHEME,
+    SpanAttributes.HTTP_URL: SpanAttributes.URL_FULL,
+    SpanAttributes.HTTP_REQUEST_CONTENT_LENGTH: SpanAttributes.HTTP_REQUEST_BODY_SIZE,
+    SpanAttributes.HTTP_RESPONSE_CONTENT_LENGTH: SpanAttributes.HTTP_RESPONSE_BODY_SIZE,
+    SpanAttributes.NET_HOST_NAME: SpanAttributes.SERVER_ADDRESS,
+    SpanAttributes.NET_HOST_PORT: SpanAttributes.SERVER_PORT,
+    SpanAttributes.NET_SOCK_HOST_ADDR: SpanAttributes.SERVER_SOCKET_ADDRESS,
```

Comment:
will need to change to `SERVER_SOCKET_ADDRESS|PORT` to `network.peer.address|port`, bit it's minor

Reply:
Doesn't this conflict with this commit? It seems net.* was changed to server.* and client.* Was it changed back to network.peer and network.local?
https://github.com/open-telemetry/semantic-conventions/commit/5b8861e8c54bc42593952faca3b9b8332cfb2d22

Reply:
the schema transformation does not capture all the changes (some of them cannot be expressed in transformation), so it's best to use [changelog](https://github.com/open-telemetry/semantic-conventions/blob/main/CHANGELOG.md) or [the spec](https://github.com/open-telemetry/semantic-conventions/blob/main/docs/http/http-spans.md) as a source of truth.

In any case, this change seems trivial.

Reply:
Got it. It seems this was just misesd because it was a newer change. The [instances of server.socket.address ](https://github.com/open-telemetry/semantic-conventions/blob/main/docs/http/http-spans.md?plain=1#L429)in the current sem conv are just mistakes. Will make this change

Reply:
Done in latest push